src/tdastro/effects/effect_model.py

```diff
@@ -0,0 +1,59 @@
+"""The base EffectModel class used for all effects."""
+
+
+class EffectModel:
+    """A physical or systematic effect to apply to an observation.
+
+    Effects are not ParameterizedNodes by can have arguments that are
+    ParameterizedNodes. All settable parameters of the EffectModel
+    must be passed as keyword arguments to the apply() method.
+
+    Attributes
+    ----------
+    rest_frame : bool
+        Whether the effect is applied in the rest frame of the observation (True)
+        or in the observed frame (False).
+    parameters : dict
+        A dictionary of parameters for the effect. Mapps parameter names to
+        their setters.
+    """
+
+    def __init__(self, rest_frame=True, **kwargs):
+        self.rest_frame = rest_frame
+
+        self.parameters = {}
+        for key, value in kwargs.items():
+            self.add_effect_parameter(key, value)
+
+    def add_effect_parameter(self, name, setter):
+        """Add a parameter to the effect.
+
+        Parameters
+        ----------
+        name : str
+            The name of the parameter.
+        setter : function
+            A function that sets the parameter value.
+        """
+        self.parameters[name] = setter
+
+    def apply(self, flux_density, rng_info=None, **kwargs):
+        """Apply the effect to observations (flux_density values)
```

Comment:
```suggestion
        """Apply the effect to observations (flux_density values).
```
(tiny tiny nit but I saw periods pointed out in a DM PR recently)